library/src/main/java/com/novoda/downloadmanager/lib/Query.java

```diff
@@ -220,14 +220,14 @@ private String getWhereClauseForBatchIds(long[] ids) {
     }
 
     private void filterByNotificationExtras(List<String> selectionParts) {
-        if (filterNotificiationExtras == null) {
+        if (filterNotificiationExtras == null || filterNotificiationExtras.length == 0) {
             return;
         }
         List<String> parts = new ArrayList<>();
         for (String filterExtra : filterNotificiationExtras) {
             parts.add(notificationExtrasClause(filterExtra));
         }
-        selectionParts.add(joinStrings(" OR ", parts));
+        selectionParts.add("(" + joinStrings(" OR ", parts) + ")");
```

Comment:
Due to the fact that at the end of all filtering in [`runQuery()`](https://github.com/novoda/download-manager/blob/master/library/src/main/java/com/novoda/downloadmanager/lib/Query.java#L186) we append an `AND deleted!=1` these `OR`s should be inclosed within brackets.